src/test/groovy/ee/tuleva/onboarding/auth/mobileid/MobileIdAuthServiceSpec.groovy

```diff
@@ -1,33 +1,192 @@
 package ee.tuleva.onboarding.auth.mobileid
 
-import com.codeborne.security.mobileid.MobileIDAuthenticator
-import com.codeborne.security.mobileid.MobileIDSession
+import ee.sk.mid.MidAuthentication
+import ee.sk.mid.MidAuthenticationResponseValidator
+import ee.sk.mid.MidClient
+import ee.sk.mid.exception.*
+import ee.sk.mid.rest.MidConnector
+import ee.sk.mid.rest.MidSessionStatusPoller
+import ee.sk.mid.rest.dao.MidSessionStatus
+import ee.sk.mid.rest.dao.response.MidAuthenticationResponse
+import ee.tuleva.onboarding.auth.exception.MobileIdException
 import spock.lang.Specification
 
 class MobileIdAuthServiceSpec extends Specification {
 
     MobileIdAuthService mobileIdAuthService
-    MobileIDAuthenticator mobileIDAuthenticator = Mock(MobileIDAuthenticator)
+    MidClient client = Mock(MidClient)
+    MidAuthenticationResponseValidator validator = Mock(MidAuthenticationResponseValidator)
 
     def setup() {
-        mobileIdAuthService = new MobileIdAuthService(mobileIDAuthenticator)
+        mobileIdAuthService = new MobileIdAuthService(client, validator)
     }
 
     def "StartLogin: Start mobile id login with a phone number"() {
         given:
-        1 * mobileIDAuthenticator.startLogin(_ as String) >> MobileIdFixture.sampleMobileIdSession
+        1 * client.getMobileIdConnector() >> Mock(MidConnector, {
```

Comment:
I think it would make sense to make the `MidConnector` and the `MidSessionStatusPoller` instances Spring Beans, so you could directly inject them into the `MobileIdAuthService` and easily mock them. 
So instead of doing:
```
1 * client.getMobileIdConnector() >> Mock(MidConnector, {
    authenticate(_) >> new MidAuthenticationResponse(MobileIdFixture.sampleSessionId)
})
```
you could do:
```
MidConnector connector = Mock(MidConnector)

...

1 * connector.authenticate(_) >> new MidAuthenticationResponse(sampleSessionId)
```